Code/GraphMol/SmilesParse/smatest.cpp

```diff
@@ -2671,11 +2671,42 @@ void testGithub2142() {
 
   BOOST_LOG(rdInfoLog) << "done" << std::endl;
 }
+
+void testGithub2565() {
+  BOOST_LOG(rdInfoLog) << "-------------------------------------" << std::endl;
+  BOOST_LOG(rdInfoLog)
+      << "Testing Github #2565: Chirality reversed on SMARTS generation."
+      << std::endl;
+
+  std::vector<std::string> smiles(
+      {R"(O=C1C[C@]([H])1F)", R"(Cl[C@@H]1CCC1=O)",
+       R"(N1CCN=C1[C@H]1CCCc2ccccc21)", R"(C[C@@H](Cl)[C@H]1CC[C@@H](Cl)CC1)",
+       R"(Fc1cn([C@@H]2CCCO2)c(=O)[nH]c1=O)",
+       R"([C@@]1(C)(C(C)(C)C)O[C@@H](CN)[C@H](C[NH3+])O1)",
+
+       // these are Ok
+       R"(O=C1C[C@](Cl)1F)", R"(Br[C@@H](Cl)F)"});
+
+  for (const auto &smi : smiles) {
+    auto *mol = SmilesToMol(smi);
```

Comment:
This line and line 2693 are going to leak core. It'd be better to use `std::unique_ptr` here

Reply:
Totally right. I was only thinking about chiralities and matches, and just missed the pointer there. I'll fix it.